controller/src/test/java/cern/jarrace/controller/rest/controller/AgentContainerControllerTest.java

```diff
@@ -0,0 +1,184 @@
+package cern.jarrace.controller.rest.controller;
+
+import cern.jarrace.controller.domain.Entrypoint;
+import org.hamcrest.core.IsEqual;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.ResultActions;
+import org.springframework.test.web.servlet.request.MockHttpServletRequestBuilder;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.util.NestedServletException;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.nio.file.Path;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasSize;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.request;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+
+/**
+ * Created by timartin on 25/01/2016.
+ */
+
+public class AgentContainerControllerTest {
+
+    private MockMvc mockMvc;
+    private AgentContainerController agentContainerController;
+
+    @Before
+    public void setUp() throws Exception {
+        agentContainerController = new AgentContainerController();
+        mockMvc = MockMvcBuilders.standaloneSetup(agentContainerController).build();
+    }
+
+    @Test
+    public void testDeployWithDifferentRequestTypes() throws Exception {
```

Comment:
consider using JUnitParams for this one. If the test fails, will you know exactly why in the traces+messages?